common/quotas/rate_limiter_impl.go

```diff
@@ -47,73 +48,54 @@ var _ RateLimiter = (*RateLimiterImpl)(nil)
 // NewRateLimiter returns a new rate limiter that can handle dynamic
 // configuration updates
 func NewRateLimiter(newRate float64, newBurst int) *RateLimiterImpl {
+	limiter := rate.NewLimiter(rate.Limit(newRate), newBurst)
+	clock := clockwork.NewRealClock()
 	rl := &RateLimiterImpl{
-		rate:          newRate,
-		burst:         newBurst,
-		goRateLimiter: rate.NewLimiter(rate.Limit(newRate), newBurst),
+		rps:                newRate,
+		burst:              newBurst,
+		clock:              clock,
+		ClockedRateLimiter: NewClockedRateLimiter(limiter, clock),
 	}
 
 	return rl
 }
 
 // SetRate set the rate of the rate limiter
-func (rl *RateLimiterImpl) SetRate(rate float64) {
-	rl.refreshInternalRateLimiterImpl(&rate, nil)
+func (rl *RateLimiterImpl) SetRate(rps float64) {
+	rl.refreshInternalRateLimiterImpl(&rps, nil)
 }
 
 // SetBurst set the burst of the rate limiter
 func (rl *RateLimiterImpl) SetBurst(burst int) {
 	rl.refreshInternalRateLimiterImpl(nil, &burst)
 }
 
-// SetRateBurst set the rate & burst of the rate limiter
-func (rl *RateLimiterImpl) SetRateBurst(rate float64, burst int) {
-	rl.refreshInternalRateLimiterImpl(&rate, &burst)
-}
-
-// Allow immediately returns with true or false indicating if a rate limit
-// token is available or not
-func (rl *RateLimiterImpl) Allow() bool {
-	return rl.goRateLimiter.Allow()
-}
-
-// AllowN immediately returns with true or false indicating if n rate limit
-// token is available or not
-func (rl *RateLimiterImpl) AllowN(now time.Time, numToken int) bool {
-	return rl.goRateLimiter.AllowN(now, numToken)
-}
-
-// Reserve reserves a rate limit token
 func (rl *RateLimiterImpl) Reserve() Reservation {
-	return rl.goRateLimiter.Reserve()
+	return rl.ClockedRateLimiter.Reserve()
 }
 
-// ReserveN reserves n rate limit token
-func (rl *RateLimiterImpl) ReserveN(now time.Time, numToken int) Reservation {
-	return rl.goRateLimiter.ReserveN(now, numToken)
+func (rl *RateLimiterImpl) ReserveN(now time.Time, n int) Reservation {
+	return rl.ClockedRateLimiter.ReserveN(now, n)
 }
```

Comment:
nit: Reserve methods can be remove as well?

Reply:
Actually, it needs to be kept because the return type of ClockedRateLimiter.Reserve* is ClockedReservation--not Reservation